JP-4285: Cube Build updates (#10433)
spacetelescope/jwst: docs/jwst/cube_build/arguments.rst

Reviewer: To make it clear you're referring to the parameters here, not the string values:
```suggestion
  with the options above (``band``, ``channel``, ``grating``, ``filter``) to fully control the type of IFU
```

  ``cube_build``, it sets ``pipeline=2``. When running stand-alone or in

>   :ref:`calwebb_spec3 <calwebb_spec3>` pipeline, the parameter ``pipeline=3`` is set for ``cube_build``.

Reviewer: Looks like this section still needs to be removed.

There are also some comments below about the "pipeline rules" for output_type that I think need to be updated.

Author: @melanieclarke 
So are your referring to these lines:
 The default rules for creating IFU cube depend on the instrument and which pipeline called ``cube_build``.

  - :ref:`calwebb_spec2 <calwebb_spec2>` pipeline rules for NIRSpec is to produce ``output_type=band``.
  - :ref:`calwebb_spec3 <calwebb_spec3>` pipeline rules for NIRSpec is to produce ``output_type=band``.

  - :ref:`calwebb_spec2 <calwebb_spec2>` pipeline rules for MIRI is to produce ``output_type=multi``.
  - :ref:`calwebb_spec3 <calwebb_spec3>` pipeline rules for MIRI is to produce ``output_type=band``.

The default output_type set in cube_build is 'band'
In calspec2, when cube_build is called  the output_type is set  for Nirspec to 'band' and MIRI as 'multi'
Did you want me to clarify that process ?

Reviewer: Yes, sorry, this was two separate comments:
1. This whole section referencing the `pipeline` parameter needs to be removed. 
2. Referring to the lines you mentioned in the `output_type` section, I think we should clarify that the default parameters in CRDS set output_type=multi for MIRI in calwebb_spec2 and otherwise, the default is "band'.

Author: To be clear the "default parameters in CRDS" do not (currently)  set the default output type. In cube_build the default output_type = 'band' . This is overridden by the calspec_spec2 pipeline for MIRI and NIRSpec

if exp_type == "NRS_IFU" and self.cube_build.output_type is None:
    self.cube_build.output_type = "band"
if is_nrs_ifu_linelamp(calibrated) and self.cube_build.output_type is None:
   self.cube_build.output_type = "band"
if exp_type == "MIR_MRS" and self.cube_build.output_type is None:
  self.cube_build.output_type = "multi"

This could easily be changed a parameter reference file.

Reviewer: This PR is changing the software default for CubeBuildStep to "band" instead of None, so this clause will no longer be hit by default.  Aren't we still planning to hold off on merging this PR until MIRI delivers a parameter reference file that selects "multi" for spec2?

``grating [string]``
  This is a NIRSpec only option with valid values PRISM, G140M, G140H, G235M, G235H, G395M, G395H, and ALL.
  If the option "ALL" is used, then all the gratings in the association are used.
  Because association tables only contain exposures of the same resolution, the use of "ALL" will at most combine
  data from gratings G140M, G235M, and G395M, or G140H, G235H, and G395H. The user can supply a comma-separated string
  containing the names of multiple gratings to use.

``filter [string]``
  This is a NIRSpec only option with values of Clear, F100LP, F070LP, F170LP, F290LP, and ALL.
  To cover the full wavelength range of NIRSpec, the option "ALL" can be used (provided the exposures in the
  association table contain all the filters). The user can supply a comma-separated string containing the names of

>   multiple filters to use.

Reviewer 2: Is this also failing to do the expected thing, in addition to channel?  This one isn't `option` type, so it should at least be possible to pass in a string like "filter1, filter2".  But maybe that breaks the code - can you confirm that such syntax is not expected to work?

Reviewer 2: @jemorrison all my comments except this one have been resolved. What are your thoughts here?

Author: I don't think I fully understand the question. How is filter different than channel, band or grating. They are all set up the same way. So if channel could not accept a list why would filter ?

 channel = option('1','2','3','4','all',default='all') # Channel
  band = option('short','medium','long','short-medium','short-long','medium-short', \
                'medium-long', 'long-short', 'long-medium','all',default='all') # Band
  grating   = option('prism','g140m','g140h','g235m','g235h','g395m','g395h','all',default='all') # Grating
  filter   = option('clear','f100lp','f070lp','f170lp','f290lp','all',default='all') # Filter

Reviewer 2: sorry I think I misread the docstring

>   This parameter has four valid options of Band, Channel, Grating, and Multi. This parameter can be combined

Reviewer: For clarity, can you please change this to the actual string values that are expected?

```suggestion
  This parameter has four valid options: "band", "channel", "grating", and "multi". This parameter can be combined
```

``grating [string]``
  This is a NIRSpec only option with valid values PRISM, G140M, G140H, G235M, G235H, G395M, G395H, and ALL.
  If the option "ALL" is used, then all the gratings in the association are used.

``filter [string]``
  This is a NIRSpec only option with values of Clear, F100LP, F070LP, F170LP, F290LP, and ALL.
  To cover the full wavelength range of NIRSpec, the option "ALL" can be used (provided the exposures in the
  association table contain all the filters). 

``output_type [string]``

>   This parameter has four valid options of "band", "channel", "grating', and 'multi". This parameter can be combined

Reviewer: One more typo here:
```suggestion
  This parameter has four valid options of "band", "channel", "grating", and "multi". This parameter can be combined
```
